it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,363 @@
+                                .setBackfillAll(Stream.BackfillAllStrategy.getDefaultInstance())
+                                .build())
+                        .build();
+
+        Stream reference = datastreamClient.createStreamAsync(request).get();
+        createdStreamIds.add(streamId);
+
+        return reference;
+    }
+
+    public Stream updateStreamState(String streamId, Stream.State state)
+            throws ExecutionException, InterruptedException {
+        Stream.Builder streamBuilder = Stream.newBuilder()
+                .setName(StreamName.format(projectId, location, streamId))
+                .setState(state);
+
+        FieldMask.Builder fieldMaskBuilder = FieldMask.newBuilder().addPaths(FIELD_STATE);
+
+        UpdateStreamRequest request = UpdateStreamRequest.newBuilder()
+                .setStream(streamBuilder)
+                .setUpdateMask(fieldMaskBuilder)
+                .build();
+
+        return datastreamClient.updateStreamAsync(request).get();
+    }
+
+    @Override
+    public Stream startStream(String streamId) throws ExecutionException, InterruptedException {
+        return updateStreamState(streamId, Stream.State.RUNNING);
+    }
+
+    @Override
+    public Stream pauseStream(String streamID) throws ExecutionException, InterruptedException {
+        return updateStreamState(streamID, Stream.State.PAUSED);
+    }
+
+    @Override
+    public void cleanupAll() {
+        try {
+            for (String stream : createdStreamIds) {
+                datastreamClient.deleteStreamAsync(DeleteStreamRequest
+                        .newBuilder()
+                        .setName(StreamName.format(projectId, location, stream))
+                        .build()).get();
+            }
+
+            for (String connectionProfile : createdConnectionProfileIds) {
+                datastreamClient.deleteConnectionProfileAsync(DeleteConnectionProfileRequest
+                        .newBuilder().setName(ConnectionProfileName.format(projectId, location, connectionProfile))
+                        .build()).get();
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    /**
+     * Builder for {@link DefaultDatastreamResourceManager}.
+     */
+    public static final class Builder {
+        private final String projectId;
+        private final String location;
+        private DatastreamClient datastreamClient;
+        private SourceType sourceType;
+        private JDBCSource jdbcSource;
+
+        private Builder(String projectId, String location) {
+            this.projectId = projectId;
+            this.location = location;
+        }
+
+        public Builder datastreamClient(DatastreamClient datastreamClient) {
+            this.datastreamClient = datastreamClient;
+            return this;
+        }
```

Comment:
The datastreamClient should be getting initialized in the constructor for the resource manager. It should never be passed in to the builder, so this setter should be removed. If there are variables that should be configured in the client, then those should have setters that the constructor can pull values from. For example, I assume there needs to be some sort of credential passed to the client similar to BigQuery and other RM's, so that should have a setter here.